expression/expression.go

```diff
@@ -85,6 +87,102 @@ type Expression interface {
 	ExplainInfo() string
 }
 
+// EvalExprsToChunk evaluates a list of expressions and append their results to "output" Chunk.
+// TODO: Determine whether some of the expressions have internal depencies and implement a vectorized execution method if not.
+func EvalExprsToChunk(ctx context.Context, exprs []Expression, input, output *chunk.Chunk) error {
+	sc := ctx.GetSessionVars().StmtCtx
+	for rowID, length := 0, input.NumRows(); rowID < length; rowID++ {
+		for colID, expr := range exprs {
+			err := evalOneCell(sc, expr, input, output, rowID, colID)
+			if err != nil {
+				return errors.Trace(err)
+			}
+		}
+	}
+	return nil
+}
+
+func evalOneCell(sc *variable.StatementContext, expr Expression, input, output *chunk.Chunk, rowID, colID int) error {
+	switch fieldType, evalType := expr.GetType(), expr.GetType().EvalType(); evalType {
```

Comment:
The `EvalType` is constant for every row, so we can loop all the rows under a certain eval type case.